terraform/service-principal.tf

```diff
@@ -0,0 +1,32 @@
+resource "azuread_application" "automation_account" {
+  display_name = "${var.automation_account_name}-automation-account"
+}
+
+resource "azuread_application_certificate" "automation_account" {
+  application_object_id = azuread_application.automation_account.id
+  type                  = "AsymmetricX509Cert"
+  encoding              = "hex"
+  value                 = azurerm_key_vault_certificate.automation_account.certificate_data
+  end_date              = azurerm_key_vault_certificate.automation_account.certificate_attribute[0].expires
+
+}
+
+resource "azuread_service_principal" "automation_account" {
+  application_id = azuread_application.automation_account.application_id
+
+  depends_on = [azuread_application_certificate.automation_account]
+}
+
+resource "azuread_service_principal_certificate" "automation_account" {
+  service_principal_id = azuread_service_principal.automation_account.id
+  type                 = "AsymmetricX509Cert"
+  encoding             = "hex"
+  value                = azurerm_key_vault_certificate.automation_account.certificate_data
+  end_date             = azurerm_key_vault_certificate.automation_account.certificate_attribute[0].expires
+}
+
+resource "azurerm_role_assignment" "key_vault" {
+  scope                = data.azurerm_key_vault.cert_key_vault.id
+  role_definition_name = "Key Vault Secrets User"
```

Comment:
does this mean you had to swap the access policy method to role from access policy on this vault

Reply:
No reason for using the role assignment, access policy slipped my mind. Is the access policy method preferred, I can change it if so.

Reply:
the role assignment method is preview and as far as I know you can't mix and match the two, we use access policies everywhere

Reply:
done